File: src/src_platform_jfx/fmg/jfx/StaticInitializer.java
package fmg.jfx;

import fmg.common.ui.Factory;
import fmg.jfx.draw.img.Animator;
import fmg.jfx.utils.Timer;

public final class StaticInitializer {

   static {
      Factory.DEFERR_INVOKER = javafx.application.Platform::runLater;
      Factory.GET_ANIMATOR = Animator::getSingleton;
      Factory.TIMER_CREATOR = Timer::new;
   }

   public static void init() {
      // implicit call static block
   }

}

File: src/src_platform_ava/fmg/ava/utils/StaticInitializer.cs
using Avalonia.Threading;
using fmg.common.ui;
using fmg.ava.img;

namespace fmg.ava.utils {

    static class StaticInitializer {

        static StaticInitializer() {
            Factory.DEFERR_INVOKER = doRun => Dispatcher.UIThread.InvokeAsync(() => doRun(), DispatcherPriority.Normal);
            Factory.GET_ANIMATOR = () => Animator.Singleton;
            Factory.TIMER_CREATOR = () => new Timer();
        }

        public static void Init() {
            // implicit call static block
        }

    }

}
